### PR TITLE
User and Channel update event handled

### PR DESCRIPTION
**Pull request by juliepromact**
User and Channel update events handled and update database accordingly. Fixed issue #77 .

**Files changed** are:
1. **IOAuthLoginRepository.cs** - Comments added and unnecessary usings removed.
2. **OAuthLoginRepository.cs** - SlackEventUpdate and SlackChannelAdd methods updated.
3. **ISlackUserRepository.cs** - UpdateSlackUser method added.
4. **SlackUserRepository.cs** - UpdateSlackUser method implemented,
5. **OAuthController.cs** - Handled "channel_rename","user_change" and "group_rename" events.

### CODE REVIEW

File: Slack.Automation/Promact.Core.Repository/SlackUserRepository/SlackUserRepository.cs [hunk 1]
```diff
@@ -29,6 +29,40 @@ public void AddSlackUser(SlackUserDetails slackUserDetails)
...
+        /// Method to update slack user 
+        /// </summary>
+        /// <param name="slackUserDetails"></param>
+        public void UpdateSlackUser(SlackUserDetails slackUserDetails)
+        {
+            var user = _slackUserDetails.FirstOrDefault(x => x.UserId == slackUserDetails.UserId);
+            if (user != null)
+            {
+                user.Deleted = slackUserDetails.Deleted;
+                user.IsAdmin = slackUserDetails.IsAdmin;
```
Comment (chintans) [inline Slack.Automation/Promact.Core.Repository/SlackUserRepository/SlackUserRepository.cs]: why have you not used automapper for assigning values to user class?

File: Slack.Automation/Promact.Core.Repository/SlackUserRepository/SlackUserRepository.cs [hunk 2]
```diff
@@ -29,6 +29,40 @@ public void AddSlackUser(SlackUserDetails slackUserDetails)
         }
 
         /// <summary>
+        /// Method to update slack user 
+        /// </summary>
+        /// <param name="slackUserDetails"></param>
+        public void UpdateSlackUser(SlackUserDetails slackUserDetails)
+        {
+            var user = _slackUserDetails.FirstOrDefault(x => x.UserId == slackUserDetails.UserId);
```
Comment (chintans) [inline Slack.Automation/Promact.Core.Repository/SlackUserRepository/SlackUserRepository.cs:64]: What happens if User is not found? How will you provide feedback to end user?

File: Slack.Automation/Promact.Erp.Core/Controllers/OAuthController.cs [hunk 3]
```diff
@@ -52,7 +64,7 @@ public IHttpActionResult RefreshToken(string refreshToken)
             }
             catch (Exception ex)
             {
-                var errorMessage = string.Format("{0}. Error -> {1}", StringConstant.LoggerErrorMessageOAuthControllerRefreshToken, ex.ToString());
+                var errorMessage = string.Format("{0}. Error -> {1}", _stringConstant.LoggerErrorMessageOAuthControllerRefreshToken, ex.ToString());
```
Comment (chintans) [inline Slack.Automation/Promact.Erp.Core/Controllers/OAuthController.cs]: Why have you used so many try catch in Refresh token method? You are catching a generic exception which we don't do.

File: Slack.Automation/Promact.Erp.Core/Controllers/OAuthController.cs [hunk 4]
```diff
@@ -119,28 +131,45 @@ public IHttpActionResult SlackEvent(SlackEventApiAC slackEvent)
...
+                        eventQueue.Dequeue();
+                        return BadRequest();
+                    }
                 }
+                return null;
             }
             catch (Exception ex)
             {
-                var errorMessage = string.Format("{0}. Error -> {1}", StringConstant.LoggerErrorMessageOAuthControllerSlackEvent, ex.ToString());
+                var errorMessage = string.Format("{0}. Error -> {1}", _stringConstant.LoggerErrorMessageOAuthControllerSlackEvent, ex.ToString());
```
Comment (chintans) [inline Slack.Automation/Promact.Erp.Core/Controllers/OAuthController.cs:169]: here also you are catching a generic exception

**Review by chintans — CHANGES_REQUESTED** [review]

File: Slack.Automation/Promact.Core.Repository/SlackUserRepository/SlackUserRepository.cs (hunk 2 above)
Comment (juliepromact) [reply Slack.Automation/Promact.Core.Repository/SlackUserRepository/SlackUserRepository.cs:64]: We can throw an exception

File: Slack.Automation/Promact.Erp.Core/Controllers/OAuthController.cs (hunk 3 above)
Comment (juliepromact) [reply Slack.Automation/Promact.Erp.Core/Controllers/OAuthController.cs]: Will change it.

File: Slack.Automation/Promact.Erp.Core/Controllers/OAuthController.cs (hunk 4 above)
Comment (juliepromact) [reply Slack.Automation/Promact.Erp.Core/Controllers/OAuthController.cs:169]: Will change it.

File: Slack.Automation/Promact.Core.Repository/SlackUserRepository/SlackUserRepository.cs (hunk 1 above)
Comment (juliepromact) [inline Slack.Automation/Promact.Core.Repository/SlackUserRepository/SlackUserRepository.cs]: Object structure required is different.

**Review by chintans — APPROVED** [review]

**Pull request merged**